themes/default/content/blog/typescript-versions/index.md

```diff
@@ -0,0 +1,97 @@
+
+Until now, Pulumi's seamless experience was limited to using Pulumi's builtin TypeScript version 3.8. If you wanted to use a more recent version, you had to introduce a build step to your project and let Pulumi run in [plain Javascript mode](https://www.pulumi.com/docs/languages-sdks/javascript/#disabling-built-in-typescript-support). With the release of the [Pulumi Node.js SDK 3.113.0](https://www.npmjs.com/package/@pulumi/pulumi) you can now use any version of TypeScript by adding it as a dependency to your project's `package.json` file.
+
+To upgrade your project to the latest versions of TypeScript and Pulumi, run:
+
+```bash
+npm add typescript@^5.4.5 @pulumi/pulumi@^3.113.0
+```
+
+Adding TypeScript as a dependency to your project is optional, TypeScript and ts-node are optional peer dependencies of the Node.js SDK. This makes it easy to use a single version of TypeScript or ts-node within monorepos. For backwards compatibility Pulumi ships with bundled versions of TypeScript 3.8 and ts-node 7, and falls back to these versions if they are not installed in your project's node modules.
+
+## Example
+
+Let's look at an example! TypeScript 4.1 introduced [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html), which allow you to build types using template literals. Imagine you wanted to enforce that names of S3 buckets are all lower case and always start with a given prefix. Of course you can check this at runtime, but with Template Literal Types this can be enforced during typechecking.
+
+```typescript
+type prefix = `acme_corp`
+type BucketName = Lowercase<`${prefix}_${string}`>;
+```
+
+We can create a custom Bucket subclass that expects a name matching our new `BucketName` type.
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+type prefix = `acme_corp`
+type BucketName = Lowercase<`${prefix}_${string}`>;
+
+class AcmeCorpBucket extends aws.s3.Bucket {
+    constructor(name: BucketName, args?: aws.s3.BucketArgs, opts?: pulumi.CustomResourceOptions) {
+        super(name, args, opts);
+    }
+}
+
+const myBucketWrong = new AcmeCorpBucket("MyBucket");
+// ❌ Fails typechecking with: Argument of type '"MyBucket"' is not assignable to parameter of type '`acme_corp_${Lowercase<string>}`'.ts(2345)
+
+const myBucketWrongAgain = new AcmeCorpBucket("acme_corp_MyBucket");
+// ❌ Fails typechecking with: Argument of type '"acme_corp_MyBucket"' is not assignable to parameter of type '`acme_corp_${Lowercase<string>}`'.ts(2345)
+
+const myBucketCorrect = new AcmeCorpBucket("acme_corp_my_bucket");
+// ✅ Passes typechecking
+```
+
+This example shows just one of the [many features](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-4.html) TypeScript has shipped since version 3.8, and now all are available for you to use in your Pulumi programs without jumping through hoops.
```

Comment:
How about the keyword link being "Typescript X version's new features" vs just "many features", where possible links should map to searchable keywords for SEO benefit.